homeassistant/components/risco/__init__.py

```diff
@@ -20,6 +20,7 @@
 
 from .const import DATA_COORDINATOR, DEFAULT_SCAN_INTERVAL, DOMAIN, EVENTS_COORDINATOR
 
+# binary_sensors need to be before sensor, to ensure the zone dict is available
```

Comment:
You don't get this guarantee? Platforms are initialized async / parallel. So in the end, this order doesn't matter and is not honored.

Reply:
Maybe I misunderstood, but I was under the impression that if you don't have any `await` in `async_setup_entry`, it'll complete before the next platform is loaded.

Reply:
It creates tasks:

```py
    for component in PLATFORMS:
        hass.async_create_task(
            hass.config_entries.async_forward_entry_setup(entry, component)
        )
```

There is no guarantee on the order

Reply:
If `async_create_task` preserves the insertion order, which I believe it does, then since they're all scheduled on the same loop, if there's no `await`, they would run in order.
From what I understand, if there are `awaits`, there's no guarantee on the order in which they'll end, but there's a guarantee on the order in which they'll start.

Reply:
but, actually there are awaits in `async_forward_entry_setup`. I'll have to change this.

Reply:
Thanks @frenck. I changed this so there's no dependency on order.